docs/cryptodoc/src/05_07_spx.rst

```diff
@@ -199,183 +207,211 @@ signing one.
-      (see :ref:`SPHINCS+ XMSS <signatures/sphincsplus/XMSS>`).
-   3. | ``SK = {secret_seed, prf, public_seed, sphincs_root}``
-      | ``PK = {public_seed, sphincs_root}``
+   1. Generate new values ``sk_seed``, ``sk_prf``, and ``pub_seed`` using ``rng``.
+   2. ``root = xmss_gen_root(secret_seed)``
+      (see :ref:`XMSS <signatures/sphincsplus/XMSS>`).
+   3. | ``SK = {sk_seed, sk_prf, pub_seed, root}``
+      | ``PK = {pub_seed, root}``
 
    **Notes:**
 
    - The creation of a public key is conducted using the
      ``public_key`` method of the private key.
+   - The addresses are set according to Algorithm 18 of [FIPS-205]_.
+
+SPHINCS\ :sup:`+`
+^^^^^^^^^^^^^^^^^
+
+Botan supports the SPHINCS\ :sup:`+` Round 3.1 NIST submission [SPX-R3]_. The
+SPHINCS\ :sup:`+` instances are activated using the ``sphincsplus_sha2`` and
+``sphincsplus_shake`` modules, enabling their selection for key creation. These
+instances are maintained solely for version compatibility. It is strongly
+recommended to use the SLH-DSA instances instead.
 
 Signature Creation
 ------------------
 
-A SPHINCS\ :sup:`+` signature is created in the following manner, following
-Algorithm 20 of [SPX-R3]_ (see :srcref:`[src/lib/pubkey/sphincsplus/sphincsplus_common]/sphincsplus.cpp:295|sign`):
+**Remark:** Signature creation with non-empty contexts is currently not
+supported in Botan. Support for the pre-hash variant of SLH-DSA is also not yet
+available.
+
+An SLH-DSA signature is created in the following manner, following
+Algorithm 22 of [FIPS-205]_ (see :srcref:`[src/lib/pubkey/sphincsplus/sphincsplus_common]/sphincsplus.cpp:355|sign`):
 
-.. admonition:: SPHINCS+ Signature Creation
+.. admonition:: SLH-DSA Signature Creation
 
    **Input:**
 
+   -  ``rng``: random number generator
    -  ``m``: message to be signed
-   -  ``SK``: SPHINCS\ :sup:`+` secret key, ``SK = {secret_seed, prf, public_seed, sphincs_root}``
+   -  ``SK``: SLH-DSA secret key, ``SK = {sk_seed, sk_prf, pub_seed, root}``
 
    **Output:**
 
-   -  ``sig``:  SPHINCS\ :sup:`+` signature
+   -  ``sig``:  SLH-DSA signature
 
    **Steps:**
 
-   1. ``opt_rand`` is set to ``SK.public_seed``. If the scheme is randomized, ``opt_rand`` is instead set to a freshly generated random byte vector.
-   2. ``msg_random_s = PRF_msg(m, SK.prf, opt_rand)`` and append ``msg_random_s`` to ``sig``.
-   3. ``mhash || tree_idx || leaf_idx = H_msg(msg_random_s, SK.sphincs_root, m)``.
-   4. Set type of ``fors_addr`` to FORS tree, its tree to ``tree_idx``, and its keypair address to ``leaf_idx``.
-   5. ``fors_sig, fors_root = fors_sign_and_pkgen(mhash, SK.secret_seed, fors_addr)`` and append ``fors_sig`` to ``sig``.
-   6. ``ht_sig = ht_sign(fors_root, SK.secret_seed, tree_idx, leaf_idx)`` and append ``ht_sig`` to ``sig``.
+   1. Generate new value ``addrnd`` using rng. For the deterministic variant, set ``addrnd`` to ``NULL``.
+   2. ``internal_msg = 0x00 || 0x00 || m`` (contexts are currently not supported).
```

Comment:
tbh now the "meat" is swapped out of the documentation, and I don't see much value in explicitly documenting the pseudocode of the external interface. Maybe we can add the previous pseudocode as documentation of `slh_sign_internal`?

Same applies to Validation.

Reply:
I've updated the algorithm descriptions accordingly.